### ISSUE TITLE
[$500] FEATURE REQUEST: Allow navigating the list in the new group dialog with arrow keys

### ISSUE THREAD

**Issue by mvtglobally** [Expensify/App#7648]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Open app
2. Navigate to "+" > "New Group"
3. Type anyones name and add
4. Use arrow keys to navigate 

## Expected Result:
User should be able to user arrow keys 

## Actual Result:
User is unable to use arrow keys 

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platform:

Where is this issue occurring?

- Web
- Desktop App

**Version Number:** 1.1.37-0
**Reproducible in staging?:** Y 
**Reproducible in production?:** Y 
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation 
![image - 2022-02-09T011717 191](https://user-images.githubusercontent.com/43995119/153133049-fb30d546-d4f8-44a1-a0de-cfaad95ecc28.png)

**Expensify/Expensify Issue URL:**
**Issue reported by:** @puneetlath 
**Slack conversation:** https://expensify.slack.com/archives/C01GTK53T8Q/p1644356006523069

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @lschurr (`AutoAssignerTriage`), see https://stackoverflow.com/c/expensify/questions/4749 for more details.

**Comment by MelvinBot** [comment]
Triggered auto assignment to @roryabraham (`Engineering`), see https://stackoverflow.com/c/expensify/questions/4319 for more details.

**Comment by roryabraham** [comment]
I have this working on a local branch but am currently scope-creeping and making arrow navigation work on all our OptionsLists, not just NewGroup. The one that's proving trickier is `IOUConfirmationList`.

**Comment by roryabraham** [comment]
Just so you know I'm not just making stuff up, here's a draft PR with what I've got so far: https://github.com/Expensify/App/pull/7702 🙃

**Comment by roryabraham** [comment]
Got my PR done and almost ready for review. It's a bit of a beast so I want to make sure I write up _super_ thorough test / QA steps.

**Comment by MelvinBot** [comment]
@roryabraham Uh oh! This issue is overdue by 2 days. Don't forget to update your issues!

**Comment by MelvinBot** [comment]
@roryabraham Still overdue 6 days?! Let's take care of this!

**Comment by roryabraham** [comment]
Okay, ended up being a very big PR but I think there's a lot of good changes there and it's very thoroughly tested. Excited to see how the review goes.

**Comment by mallenexpensify** [comment]
Assigned to @rushatgabhane , he's reviewing and testing the PR, job price for doing so will be $500. 
Rushat, please apply here https://www.upwork.com/jobs/~016a9b6f5ca7f87207

**Comment by mallenexpensify** [comment]
PR is being reviewed, it's a bit large :)

**Comment by roryabraham** [comment]
Yeah, I'll try to push it forward ASAP. I might split it into multiple PRs to make it easier for the reviewers too. @rushatgabhane would it be helpful it I split off the KeyboardShortcut library changes from the rest?

**Comment by rushatgabhane** [comment]
@roryabraham [participants being hidden](https://github.com/Expensify/App/pull/7702#issuecomment-1058343041) is the main blocker for the PR, and  

Splitting off KeyboardShortcut lib would be nice because there are at least two issues waiting on event bubbling to be implemented.

**Comment by roryabraham** [comment]
I'm sure we'll be able to fix the participants being hidden bug, but frankly this just isn't as high priority as some other issues I've got assigned to me at the moment. So I'm going to create a separate PR for the KeyboardShortcut lib upgrades then follow-up with this as soon as I can.

**Comment by mallenexpensify** [comment]
@roryabraham should @rushatgabhane be compensated additionally for the work reviewing the new PR https://github.com/Expensify/App/pull/8059

**Comment by rushatgabhane** [comment]
For transparency, during review of original PR, I missed that bug in the first place.

**Comment by mallenexpensify** [comment]
PR was put on hold, checking to see if it can be taken off
https://github.com/Expensify/App/pull/7702#issuecomment-1076820876

**Comment by roryabraham** [comment]
It can be taken off hold, I just haven't been able to re-prioritize it yet. I could clean it up and get it ready for review then we could treat the known bug (where the first item is not always visible) as a separate bug and make it external. Thoughts @rushatgabhane?

**Comment by rushatgabhane** [comment]
@roryabraham sure, let's do that

**Comment by mallenexpensify** [comment]
I'm not sure where this issue is at.  Rory or Rushat, can you provide an update?

**Comment by rushatgabhane** [comment]
@mallenexpensify my best guess would be that Rory has other higher priority issues, so the PR is on a pseudo hold.

**Comment by mallenexpensify** [comment]
hahahah I love `pseudo hold`, def hoping to reuse that term

**Comment by roryabraham** [comment]
Nope, the PR has been updated recently and is in active review!

**Comment by mallenexpensify** [comment]
Based on this comment in the PR, it looks like Rory needs to review Marc's review
https://github.com/Expensify/App/pull/7702#pullrequestreview-942599372

Also... reminder to self, this is the issue Rushat was hired to review the PR, I believe the compensation is $500, will double check when issuing payment

**Comment by roryabraham** [comment]
Yeah, quick update – @marcaaron's review was pretty substantial and so I'm working on some of the refactors he asked for in a separate branch. I'm making progress, but not pushing to the branch until I have something I'm confident in.

**Comment by rushatgabhane** [comment]
@roryabraham I'm guessing the PR will require further C+ review. 
Feel free to unassign me if that's the case because I'm OOO until May 14.

**Comment by mallenexpensify** [comment]
I think @roryabraham is working on this

**Comment by mallenexpensify** [comment]
Looks like @parasharrajat is helping review the PR, which is moving along. 
Reminder to discuss potential compensation for Rushat and Rajat once this is closed

**Comment by mallenexpensify** [comment]
Not much action in the PR the past couple weeks
https://github.com/Expensify/App/pull/7702

**Comment by roryabraham** [comment]
Got the PR all cleaned up and ready for review

**Comment by roryabraham** [comment]
@rushatgabhane has been a 🦸 reviewer on this massive PR. Let's bump his compensation for this issue up [from $500](https://github.com/Expensify/App/issues/7648#issuecomment-1104140533) to $1000 (assuming no regressions).

**Comment by mallenexpensify** [comment]
Whoohoo!! Thanks for the help @rushatgabhane !    Y'all might have to let me know when to issue payment, if the title doesn't get auto-updated with the date

**Comment by mallenexpensify** [comment]
Looks like it's still being worked on and reviewed, Rory's out this week too

**Comment by mallenexpensify** [comment]
@roryabraham or @rushatgabhane , can you provide a `tl,dr:` on:
1. Why this is such a big undertaking
2. Will this help on other parts of the site for navigating via arrow keys?
(I'm not asking cuz it's taking a bit but curious who it's a lot of work vs _just_ mimicking the use of tab and shift+tab for keyboard nav)

Also.. for other keyboard nav improvements for group chats, dropped this in #expensify-open-source 
https://expensify.slack.com/archives/C01GTK53T8Q/p1655918174158159

**Comment by roryabraham** [comment]
> Will this help on other parts of the site for navigating via arrow keys?

Yes, the PR I have been working on implements arrow key navigation in many places on the site.

> Why this is such a big undertaking

Because I chose to solve this class of problem almost everywhere in the app, it involves a large refactor of some of our most complex components.

**Comment by mallenexpensify** [comment]
> Because I chose to solve this class of problem almost everywhere in the app, it involves a large refactor of some of our most complex components.

whoooohooo.. thanks @roryabraham (and @rushatgabhane ) keyboard nav and shortcuts FTW!!!

**Comment by roryabraham** [comment]
I think @rushatgabhane should get $1000 for the review on this issue and @parasharrajat gets

**Comment by rushatgabhane** [comment]
Thank you for reopening!

**Comment by mallenexpensify** [comment]
@roryabraham or @rushatgabhane what's the payment date? I want to update the title so I don't forget

**Comment by rushatgabhane** [comment]
We'll know the date after the PR is on production. 

Also, we had one regression so the payment should be $500
- https://github.com/Expensify/App/issues/9606

**Comment by rushatgabhane** [comment]
@mallenexpensify so far there has been one regression - #9606 

Rest of them aren't regressions
1. https://github.com/Expensify/App/issues/9598
2. https://github.com/Expensify/App/issues/9614
3. https://github.com/Expensify/App/issues/9604
4. https://github.com/Expensify/App/issues/9595

**Comment by rushatgabhane** [comment]
Regression # 2 - https://github.com/Expensify/App/issues/9678

**Comment by Christinadobrzyn** [comment]
Looks like old job posting in Upwork was closed - so created a new posting here. 

Internal - https://www.upwork.com/jobs/~017c647a0e75584044
External - https://www.upwork.com/ab/applicants/1544742427005378560/job-details

**Comment by rushatgabhane** [comment]
@Christinadobrzyn i guess we can settle this issue

PR merged to production ~20 days ago https://github.com/Expensify/App/pull/7702#issuecomment-1178546074

**Comment by Christinadobrzyn** [comment]
Paid @rushatgabhane $500 for the job. 

I can see that we've only hired @rushatgabhane so I assume this is the only payment? If that's wrong, please let me know. 

Closed job in Upwork and closing this as complete.

**Comment by parasharrajat** [comment]
I think I am left out https://github.com/Expensify/App/issues/7648#issuecomment-1176518162.

**Comment by Christinadobrzyn** [comment]
Oh no! I'm not sure what I did here or if this is bad - @roryabraham would you be able to lend some insight?

**Comment by Christinadobrzyn** [comment]
Not sure why this auto-message is posting so many times, asking eng - https://expensify.slack.com/archives/C03TQ48KC/p1659380948350629

**Comment by marcaaron** [comment]
On it.

**Comment by marcaaron** [comment]
Sorry about that everyone. We accidentally created a bug when adding the new reminder feature. I think it should be resolved now. Going to leave a quick test comment here to confirm that can be ignored.

Test: https://github.com/Expensify/App/issues/7648

**Comment by Christinadobrzyn** [comment]
Yes, based on this [comment](https://github.com/Expensify/App/issues/7648#issuecomment-1168078917), you should have been paid $1000 - I didn't see that at the time of paying you for this job.

**Comment by Christinadobrzyn** [comment]
Sounds good, I wasn't aware of the regressions, is it an option to reject the bonus?

**Comment by Christinadobrzyn** [comment]
Sounds good, I just requested a refund @rushatgabhane 

@parasharrajat I also created this new job to pay you the $250 - https://www.upwork.com/ab/applicants/1554909413932630016/job-details

Let me know when you've accepted the offer and I'll pay you.

**Comment by Christinadobrzyn** [comment]
Paid @parasharrajat through this job posting - https://www.upwork.com/ab/applicants/1554909413932630016/job-details

I think we're all set here! Let me know if there's anything else.
